### ISSUE TITLE
iOS GCS

### ISSUE THREAD

**Issue by tgreer** [d-ronin/dRonin#725]
Opening issue to discuss iOS GCS. There are solutions out there to create a TCP<>Serial bridge, https://github.com/beckdac/ESP8266-transparent-bridge as an example uses a $5 component.

[edit] GCS currently does TCP or UDP.

**Comment by WZ9V** [comment]
I'd be interested in this also.

**Comment by mluessi** [comment]
AFAIK, the GPL is incompatible with the app-store TOS, so the iOS GCS would need have to have a different licence and therefore would probably have to written from scratch.

**Comment by tgreer** [comment]
@mluessi The internet agrees with you. I think it'd need to be written from scratch anyways...

**Comment by mluessi** [comment]
@tgreer yeah, the license thing is unfortunate. One possibility could be to build a Qt / QML based app, which would have the advantage that it would also work on Android. In addition, we could probably license the app such that components from it could be used in the normal GCS (but not the other way around).

The question is, what is the scope of the app? Recreate the whole GCS, i.e., board configuration, calibration, etc, or do we want an app for doing navigation flights (setting waypoints etc)? The latter may be more easily accomplished by improving our MAVLINK support, so existing apps for APM could be used.

**Comment by tgreer** [comment]
I'd like it to be as featured as the existing Android one... Granted if
we're starting from scratch the remit can be whatever the dev decides... I
would like it to be fully functioning... but I can dream, right ;)

**Comment by WZ9V** [comment]
I'm willing to help but I'm currently in over my head working on a BLHELI GUI in Qt that I'm hoping can be the basis for adding that functionality to the GCS.  Keep in mind I'm also learning Qt at the same time ;-)

**Comment by sonicxx** [comment]
There's also the possibility of releasing without the App Store as an enterprise app. Also there is cydia for jailbreakers. 

If full functionality is desired, reusing the code is a good leg up. I don't see any down side, except for the sourcing and cost of an Enterprise certificate.

Also, VLC is back in the AppStore and was the major GPL example.  I think maybe they believe that because they are distributing binaries/source separately from the App Store and they include the rights on those,  it's a superset of the existing App Store license.

**Comment by sonicxx** [comment]
Apparently it had something to do with Mozilla software licensing? As to why they feel they are in full compliance.

**Comment by mlyle** [comment]
Yes, if it's licensed under a license that doesn't have the "no further restrictions" clause all is good.

Unfortunately it requires consent of license holders to do this.
